enhancements/oc/day2-add-nodes.md

```diff
@@ -0,0 +1,264 @@
+4. The cluster administrator runs the `oc adm monitor-add-nodes` command to monitor the activity progress
+5. Once a node csr is pending, the cluster administrator runs the `oc adm certificate approve <csr-id>` to complete the process for the specified node
+
+### API Extensions
+N/A
+
+### Implementation Details/Notes/Constraints
+The implementation will largely reuse the same approach / code present in the openshift/installer repo for creating an ISO image 
+for the installation, with a number of exceptions that will be detailed in the subsequent paragraphs.
+
+#### Proposed architecture
+Every node booted with the generated ISO will will be independent for adding itself to the cluster - without any need to specify a single rendezvous IP, 
+as it happens for the installation scenario. In this way if a node is failing for some reason (for example, the pre-flight validations block the process), 
+the other nodes may continue the joining process undisturbed.
+The following hilights better the proposed architecture:
+
+![alt Architecture diagram](day2-add-nodes-diagram.png)
+
+#### Assisted-service interaction
+The following API calls will be used to add a single node to the target cluster:
+- `/api/assisted-install/v2/clusters/import` to import an existing cluster
+- `/api/assisted-install/infra-envs/${INFRA_ENV_ID}/hosts/${HOST_ID}` to add the specified host to the cluster
+
+#### oc integration notes
+Since most of the existing code for agent-based installer approach is currently stored into the openshift/installer repo, the proposal is to implement the 
+commands in a way similar to the already existing `oc adm must-gather` command, where essentially a pod is run within the target cluster to execute specific 
+code build from the installer and shipped into a release image. Such approach will offer the following benefits:
+
+- No code duplication/pollution. The required code for the additional commands will exist only in the openshift/installer repo (it will not be vendored into 
+  the openshift/oc repo)
+- Simplified integration. `oc` will take care only of marshaling the inputs/outputs of the command
+
+To achieve this kind of loosely coupled level of integration with `oc`, a specific binary with the described commands will be built from the openshift/installer 
+repo code and distributed into an image of the release payload as an additional artifact.
+
+### Risks and Mitigations
+Most of the existing code for agent-based installer approach is currently stored into the openshift/installer repo (with additional vendoring of the api/client 
+parts of openshift/assisted-service), and it wasn’t explicitly designed for being re-vendored. Vendoring the required parts in the openshift/oc repo will require
+a significant amount of refactoring in the openshift/install repo. Also, `oc` is not tied to a particular version, while the agent-based installer code is. 
```

Comment:
oc has supported version skew +1/-1 guarantee and it seems that this may break that guarantee.